editor/test/effects.js

```diff
@@ -319,6 +320,118 @@ describe( 'effects', () => {
 			expect( dispatch ).toHaveBeenCalledTimes( 1 );
 			expect( dispatch ).toHaveBeenCalledWith( requestMetaBoxUpdates( [ 'normal', 'side' ] ) );
 		} );
+
+		it( 'should dispatch notices when reverting a published post to a draft.', () => {
+			const dispatch = jest.fn();
+			const store = { getState: () => {}, dispatch };
+
+			selectors.getDirtyMetaBoxes.mockReturnValue( [ 'normal', 'side' ] );
+
+			const post = {
+				id: 1,
+				title: {
+					raw: 'A History of Pork',
+				},
+				content: {
+					raw: '',
+				},
+				status: 'publish',
+			};
+
+			const previousPost = {
+				...post,
+				status: 'draft',
+			};
+
+			handler( { post, previousPost }, store );
+
+			expect( dispatch ).toHaveBeenCalledTimes( 2 );
+			expect( dispatch ).toHaveBeenCalledWith( expect.objectContaining( {
+				notice: {
+					content: <p><span>Post published!</span> <a href={ undefined }>View post</a></p>, // eslint-disable-line jsx-a11y/anchor-is-valid
+					id: 'SAVE_POST_NOTICE_ID',
+					isDismissible: true,
+					status: 'success',
+				},
+				type: 'CREATE_NOTICE',
+			} ) );
+		} );
+
+		it( 'should dispatch notices when publishing or scheduling a post.', () => {
+			const dispatch = jest.fn();
+			const store = { getState: () => {}, dispatch };
+
+			selectors.getDirtyMetaBoxes.mockReturnValue( [ 'normal', 'side' ] );
+
+			const post = {
+				id: 1,
+				title: {
+					raw: 'A History of Pork',
+				},
+				content: {
+					raw: '',
+				},
+				status: 'draft',
+			};
+
+			const previousPost = {
+				...post,
+				status: 'publish',
+			};
+
+			handler( { post, previousPost }, store );
+
+			expect( dispatch ).toHaveBeenCalledTimes( 2 );
+			expect( dispatch ).toHaveBeenCalledWith( expect.objectContaining( {
+				notice: {
+					content: <p>
+						<span>Post reverted to draft.</span>
+						{ ' ' }
+						{ false && <a href={ post.link }>{ 'View post' }</a> }
+					</p>,
+					id: 'SAVE_POST_NOTICE_ID',
+					isDismissible: true,
+					status: 'success',
+				},
+				type: 'CREATE_NOTICE',
+			} ) );
+		} );
+
+		it( 'should dispatch notices when just updating a published post again.', () => {
```

Comment:
Just noting, that we don't use a period at the end of the test name in other places.